File: Swift/프로그래머스/2/140107. 점 찍기/점 찍기.swift
import Foundation

func solution(_ k:Int, _ d:Int) -> Int64 {
    var answer: Int64 = 0
    
    let D = Double(d)
    
    var x = 0
    while x <= d {
        let dX = Double(x)
        
        let maxY = sqrt(pow(D, 2) - pow(dX, 2))
        
        answer += (Int64(maxY / Double(k)) + 1)
        x += k
    }
    
    return answer
}

File: Java/프로그래머스/2/140107. 점 찍기/점 찍기.java
class Solution {
    public long solution(int k, int d) {
        long answer = 0;
        
        for (long x = 0; x<= d; x += k) {
            long dSquared = (long) d * d;
            long xSquared = x * x;
            
            double maxY = Math.sqrt(dSquared - xSquared);
            
            answer += ((long)(maxY / k)) + 1;
        }
        return answer;
    }
}